### PR TITLE
This pull request adds support for several RegEx character classes

### PR DESCRIPTION
**Pull request by corsonknowles**
Newly included character classes: 
    // "\d" any non-digit character
    // \b	word boundary
    // \B not-word boundary
    // carriage return \r

The above comments can be removed from methodMatch.js, and are only included to make code review easier. 

Newly added:
  support for negation of a group: none of abc -> [^abc]

Removed these RegEx characters that are not supported in JavaScript
    // \A start of string, contrast with ^ which breaks at new lines
    // \z end of string, contrast with $ which breaks at new lines

### CODE REVIEW

**Comment by KarimGeiger** [comment]
Hi @corsonknowles, thanks again for this contribution. It looks good to me. 

@teabyii, what's your opinion here? Are you missing something, or can we merge those changes?

File: package.json [hunk 1]
```diff
@@ -33,5 +33,8 @@
     "istanbul": "^0.4.5",
     "mocha": "^3.0.2",
     "mocha-lcov-reporter": "^1.2.0"
+  },
+  "dependencies": {
+    "test": "^0.6.0"
```
Comment (teabyii) [inline package.json]: I don't think it needs this dependency.

**Review by teabyii — COMMENTED** [review]
Remove `package-lock.json`？

**Comment by teabyii** [comment]
Thank you for the PR, and try to increase the code coverage?

**Comment by teabyii** [comment]
Why remove submodule and add `Test-Rules` code directly to this repo? I don't think it's a good way to maintenance the common test rules for different languages.

**Comment by corsonknowles** [comment]
Ah. I think I misunderstood the requests to add tests and increase coverage
and also the comment about removing the dependency. What is the correct
approach? Karim also asked me to write rules for the new RegEx characters
supported, and I followed the readme docs for the .rule files

I would be very happy to finish this today. Ideally, we will be giving a
demonstration with SRL-JavaScript tomorrow, and it would be very nice to
get to demonstrate it with these few minor updates in place.

**Comment by corsonknowles** [comment]
Ah, I see. I have moved the .rules files to a new Pull Request for the
relevant submodule in the rules repo.

I removed the local copy of the .rules files, and the changes to the
submodule for SRL-JavaScript.

Thanks!

File: lib/Builder.js [hunk 2]
```diff
@@ -39,16 +39,31 @@ const simpleMapper = {
...
+        'add': '\\',
+        'type': METHOD_TYPE_CHARACTER,
+        'allowed': METHOD_TYPES_ALLOWED_FOR_CHARACTERS
+    },
     'tab': {
         'add': '\\t',
         'type': METHOD_TYPE_CHARACTER,
         'allowed': METHOD_TYPES_ALLOWED_FOR_CHARACTERS
     },
+    'vertical tab': {
```
Comment (KarimGeiger) [inline lib/Builder.js]: Please stick with the camelCase convention there.

**Comment by corsonknowles** [comment]
Thanks! Corrected.

**Comment by teabyii** [comment]
Well done. Now Wait for [PR for Test-rules](https://github.com/SimpleRegex/Test-Rules/pull/2) to be merged.

**Comment by corsonknowles** [comment]
Thank you! Rules just merged.

**Comment by teabyii** [comment]
@corsonknowles Reorganize the commits and put them in one commit please? That looks more beautiful, thank you.

**Comment by KarimGeiger** [comment]
@teabyii you can do this using GitHub while merging. Just select "Squash and merge" to combine them.

**Comment by teabyii** [comment]
Ok, I got it.

**Pull request merged**